precompiles/conviction-voting/src/lib.rs

```diff
@@ -85,30 +106,47 @@ where
 		vote_amount: U256,
 		conviction: u8,
 	) -> EvmResult {
-		let poll_index = Self::u32_to_index(poll_index).in_field("pollIndex")?;
-		let vote_amount = Self::u256_to_amount(vote_amount).in_field("voteAmount")?;
-		let conviction = Self::u8_to_conviction(conviction).in_field("conviction")?;
-
+		let caller = handle.context().caller;
 		let aye = match vote {
 			VoteDirection::Yes => true,
 			VoteDirection::No => false,
 			_ => return Err(RevertReason::custom("Abstain not supported").into()),
 		};
 
-		let vote = AccountVote::Standard {
-			vote: Vote { aye, conviction },
-			balance: vote_amount,
-		};
-
-		log::trace!(target: "conviction-voting-precompile",
-			"Voting {:?} on poll {:?}, with conviction {:?}",
-			aye, poll_index, conviction
-		);
-
-		let origin = Runtime::AddressMapping::into_account_id(handle.context().caller);
-		let call = ConvictionVotingCall::<Runtime>::vote { poll_index, vote }.into();
-
-		<RuntimeHelper<Runtime>>::try_dispatch(handle, Some(origin).into(), call)?;
+		{
+			let poll_index = Self::u32_to_index(poll_index).in_field("pollIndex")?;
+			let vote_amount = Self::u256_to_amount(vote_amount).in_field("voteAmount")?;
+			let conviction = Self::u8_to_conviction(conviction).in_field("conviction")?;
+
+			let vote = AccountVote::Standard {
+				vote: Vote { aye, conviction },
+				balance: vote_amount,
+			};
+
+			log::trace!(target: "conviction-voting-precompile",
+				"Voting {:?} on poll {:?}, with conviction {:?}",
+				aye, poll_index, conviction
+			);
+
+			let origin = Runtime::AddressMapping::into_account_id(caller);
+			let call = ConvictionVotingCall::<Runtime>::vote { poll_index, vote }.into();
+
+			<RuntimeHelper<Runtime>>::try_dispatch(handle, Some(origin).into(), call)?;
+		}
+
+		handle.record_log_costs_manual(2, 32 * 4)?;
```

Comment:
so it is 32 * number of arguments in the event struct? Is this documented somewhere or can we add a comment for posterity

Reply:
Maybe unnecessary, I realize now that all arguments take up 32 bytes because that is how solidity is.